### PR TITLE
Orbit Controller refactor to improve pan, rotation and zoom

### PR DESCRIPTION
**Pull request by jianhuang01**
This is the second round to improve Orbit Controller, includes these items.
1. Continue to move panning from clip space to world space, which can avoid rendering distortion and inaccurate depth clipping
2. Change the old Orbit State modeling from (lookAt, translateX, translateY, pitchAngle, rotationAngle) to (lookAt, pitchAngle, rotationAngle), cause translate and rotation is dependent with each other which makes the logic much more complicated.
3. Clean up code for pan, rotation and zoom to adapt new modeling

Tested with plot, point-cloud-laz and point-cloud

### CODE REVIEW

File: src/core/controllers/orbit-state.js [hunk 1]
```diff
@@ -50,9 +36,9 @@ export default class OrbitState {
     width, // Width of viewport
     height, // Height of viewport
     distance, // From eye to target
-    pitchAngle, // Rotation around x axis
-    orbitAngle, // Rotation around orbit axis
-
+    rotationX, // Rotation around x axis
```
Comment (gnavvy) [inline src/core/controllers/orbit-state.js]: personally, I agree with the original naming that _pitch_ is relatively easier to understand than _X_.
https://social.sbrick.com/custom/wiki/designer/latest/camera_pitch_yaw_roll.png

but as long as we have documented this clearly it should be fine.

File: src/core/controllers/orbit-state.js [hunk 2]
```diff
@@ -147,19 +136,25 @@ export default class OrbitState {
    * @param {[Number, Number]} pos - position on screen where the pointer is
    */
   pan({pos, startPos}) {
+    if (this._interactiveState.isRotating) {
```
Comment (1chandu) [inline src/core/controllers/orbit-state.js]: Don't have to address in this PR.
Quesiton: Can we make the decision of panning/rotating before pan()/rotate() gets called so we don't need these if conditions?

File: src/core/controllers/orbit-state.js [hunk 3]
```diff
@@ -195,40 +189,23 @@ export default class OrbitState {
...
-    orbitAngle = ensureFinite(orbitAngle, this._viewportProps.orbitAngle);
-    translationX = ensureFinite(translationX, this._viewportProps.translationX);
-    translationY = ensureFinite(translationY, this._viewportProps.translationY);
-
-    const newPitchAngle = clamp(pitchAngle - deltaScaleY * 180, -89.999, 89.999);
-    const newOrbitAngle = (orbitAngle - deltaScaleX * 180) % 360;
-
-    let newTranslationX = translationX;
-    let newTranslationY = translationY;
+    if (this._interactiveState.isPanning) {
```
Comment (1chandu) [inline src/core/controllers/orbit-state.js]: Same comment as above.

File: src/core/controllers/orbit-state.js (hunk 1 above)
Comment (1chandu) [reply src/core/controllers/orbit-state.js]: On this same topic, we should also document that, OrbitAxis can't be x-axis.

File: src/core/controllers/orbit-state.js (hunk 2 above)
Comment (jianhuang01) [reply src/core/controllers/orbit-state.js]: I like that, too. The logic is inside ViewControls which will impact all the viewports, and ViewContols is under refactor, need discuss before the change. That's why I didn't do that.

**Pull request merged**